app/src/main/java/com/github/wnder/TakePictureActivity.java

```diff
@@ -64,14 +76,20 @@ protected void onCreate(Bundle savedInstanceState) {
         pictureConfirmButton = findViewById(R.id.pictureConfirmButton);
         pictureConfirmButton.setVisibility(View.INVISIBLE);
         pictureConfirmButton.setOnClickListener((view) -> {
-            storeBitmapInGallery();
-
-            boolean hasSucceeded = storeBitmapInDB();
-
-            Intent intent = new Intent(this, UploadActivity.class);
-            intent.putExtra(HAS_SUCCEEDED, hasSucceeded);
-            startActivity(intent);
-            this.finish();
+            if (networkInfo.isNetworkAvailable()){
+                storeBitmapInGallery();
+
+                boolean hasSucceeded = storeBitmapInDB();
+
+                Intent intent = new Intent(this, UploadActivity.class);
+                intent.putExtra(HAS_SUCCEEDED, hasSucceeded);
+                startActivity(intent);
+                this.finish();
+            }
+            else{
+                AlertDialog alert = AlertBuilder.noConnectionAlert(getString(R.string.no_connection), getString(R.string.no_internet_body), this);
+                alert.show();
+            }
```

Comment:
Maybe we could let the user store the picture in his gallery even without connection, and only prevent him from uploading it to the db